packages/notarization/commitments.go

```diff
@@ -339,13 +339,18 @@ func (f *EpochCommitmentFactory) newEpochRoots(ei epoch.Index) (commitmentRoots
 	// We advance the LedgerState to the next epoch.
 	f.commitLedgerState(ei - epoch.Index(f.snapshotDepth))
 
-	return &CommitmentRoots{
+	commitmentRoots = &CommitmentRoots{
 		EI:                ei,
 		stateRoot:         epoch.NewMerkleRoot(stateRoot),
 		manaRoot:          epoch.NewMerkleRoot(manaRoot),
 		tangleRoot:        epoch.NewMerkleRoot(commitmentTrees.tangleTree.Root()),
 		stateMutationRoot: epoch.NewMerkleRoot(commitmentTrees.stateMutationTree.Root()),
-	}, nil
+	}
+
+	// We are never going to use this epoch's commitment trees again.
+	delete(f.commitmentTrees, ei)
```

Comment:
After this deletion, the proof of inclusion is not available anymore, should we keep the tree until pruning or epoch notarized, or even make it configurable?

Reply:
We had a few discussions about this, and we concluded that it is not the job of the core plugin to generate proofs, but rather have a specialized plugin that deals with such things by storing data for the required amount of time.
If you notice I've separated all the proofs related stuff in a separate file, and I would actually propose to drop it for the time being: still it will be nice to have some tests to actually create and verify some proofs of inclusion: mostly to test that the trees are fine rather than having a proofing / verifying framework.
What do you think @daria305 @jkrvivian ?

Reply:
I agree, then we can have a discussion about testing :tada:

Reply:
We could add the proof-related things in form of a plugin, in some of next sprints, after we finish switching to heavier chains. But for now I agree we can limit to some simpe tests with the usage of proof functions.